### ISSUE TITLE
alexnet produces incorrect results

### ISSUE THREAD

**Issue by mvermeulen** [ROCmSoftwarePlatform/AMDMIGraphX#209]
alexnet pretrained model from the torchvision model zoo has correctness issues.

Using the 50,000 imagenet validation images, the largest value is highest in one of three buckets:
- 703 - park bench
- 927 - trifle
- 962 - meat loaf, meatloaf
No other buckets are every the highest.

Note that using pytorch-cpu I can verify a "top1" score of 56.52% for the saved model itself vs the published 56.55%; so I think the ONNX file is ok.  Other models such as resnet50 do validate with MIGraphX, so something unique is going on with alexnet.

**Comment by mvermeulen** [comment]
Another datapoint: AMD ROCm pytorch container also works as expected; with a "top1" score of 56.63%

**Comment by mvermeulen** [comment]
Python script that was used to measure alexnet and other models.

[migraphx-classify.py.txt](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/files/2963483/migraphx-classify.py.txt)

Note: on my test system, I ran this script in a child directory of the imagenet validation images, hence the chdir().  Could otherwise be cleaned up, but attach it here if it helps reproduce the example.

**Comment by mvermeulen** [comment]
Documenting few observations and a some hypothesis:
1. @kahmed10 can reproduce the symptoms but he gets different wrong answers
2. ONNX files generated from PyTorch 0.4.0 and PyTorch 1.0 both reproduce the symptoms but get different wrong answers.
3. verify_onnx passes.
4. Turning off memory coloring gives the same wrong answers.
5. Trying the following
```
env MIOPEN_ENABLE_LOGGING=1 python migraphx-image.py alexnet ../ILSVRC2012_val_0000001.JPG
```
results in no calls to routines with "LRN" in their names.

Hypothesis:
1. Symptoms above suggest reading from uninitialized or incorrect memory.  Candidates might be:
- bad ONNX files - except both 1.0 and 0.4?
- padding differences?
- issues in implementation of lrn operator
- bad optimizations that create incorrect aliases
2. The last symptom at least suggest tracing steps involved in LRN implementation to ensure either that "miopen_lrn::compute" is not supposed to be called.

**Comment by mvermeulen** [comment]
Another datapoint:
- commented out all the passes in target::get_passes except: lowering and write_literals.  Alexnet still fails in same fashion.

**Comment by pfultz2** [comment]
> commented out all the passes in target::get_passes except: lowering and write_literals. Alexnet still fails in same fashion.

This would be expected. Since verify_onnx passes, then we already know there is no issue with optimization passes.

**Comment by pfultz2** [comment]
I ran the torchvision model through ubsan, asan, and gcc 7's `-fsanitize-address-use-after-scope` and saw no issue.

Next, I'll try running it through valgrind to see if I can find any issues.

I am not sure if its related, there is a known issue with using gcc 7's `-fbounds-strict` in the generated protobuf files. I haven't investigated enough to know if its a FP, a bug in the protobufs generated code, or a bug in the way we are calling protobuf.

**Comment by mvermeulen** [comment]
First pass trying ONNX files with onnx_tf backend is not 100% conclusive either way (either that there is or isn't an issue with the ONNX file.  Rather than real images, my script generates a random array (using the same seed each time) and then runs it through the network.  Using the Alexnet ONNX file, I get the same top5 indices from both TF and MIGraphX
```
530 51 523 605 397
```
I also get the same top5 indices for densenet121 and vgg16; curiously these are same between TF/MIgraphX but different from each other
```
densenet121 539 852 611 735 490
vgg16 556 980 794 921 619
```
Curious enough, I don't get the same values for resnet50
```
resnet50 MIGraphX 745 795 366 245 791
resnet50 Tensorflow 632 111 107 644 626
```
So I am mostly assuming I have other anomalies going on for resnet50 and between models using a completely randomly generated image.  I don't have enough info to know whether the alexnet computation is correct or not.  Will try some things to see if I can try the ONNX files with real images with other backends to get a better confirmation pro/con on the ONNX file itself.

By the way, the webcam demo also fails with alexnet.  This doesn't tell us anything new, but does confirm that we're not getting right results from alexnet.

**Comment by mvermeulen** [comment]
After further investigation with @kahmed10 looks like this one was user error missing a "pretrained" flag in the ONNX file generation.  Running the measurements now, but initial indications are that Alexnet works.  Once I the pass rate run is complete, anticipate will close this as a user error.

**Comment by mvermeulen** [comment]
As anticipated - results for Alexnet are now:
- top1 = 56.63% (vs. published 56.55%)
- top5 = 79.05% (vs. published 79.09%)
Closing this as user error.
